src/value/tagged.rs

```diff
@@ -260,6 +260,43 @@ impl<'de> EnumAccess<'de> for TaggedValue {
     }
 }
 
+pub struct TaggedValueMapAccess {
+    inner: TaggedValue,
+    has_visited: bool,
+}
+
+impl From<TaggedValue> for TaggedValueMapAccess {
+    fn from(inner: TaggedValue) -> Self {
+        TaggedValueMapAccess {
+            inner,
+            has_visited: false,
+        }
+    }
+}
+
+impl<'de> MapAccess<'de> for TaggedValueMapAccess {
+    type Error = Error;
+
+    fn next_key_seed<K>(&mut self, seed: K) -> Result<Option<K::Value>, Error>
+    where
+        K: DeserializeSeed<'de>,
+    {
+        if self.has_visited {
+            return Ok(None);
+        }
+        self.has_visited = true;
+        let tag = StrDeserializer::<Error>::new(nobang(&self.inner.tag.string));
+        seed.deserialize(tag).map(Some)
+    }
+
+    fn next_value_seed<V>(&mut self, seed: V) -> Result<V::Value, Error>
+    where
+        V: DeserializeSeed<'de>,
+    {
+        seed.deserialize(self.inner.value.clone())
```

Comment:
Strange, my second comment from review disappeared from it. Is that clone necessary? You can store necessary data in Option and steal it using `Option::take` here. Then `has_visited` could be replaced by `Option::is_none()`